### ISSUE TITLE
About the GAN's results, I have trouble!

### ISSUE THREAD

**Issue by dljlw** [dljlw/GAN#1]
From the picture's numbers, we can see that all numbers of columns are the same. I use the GANs' Generator to generate simulation, but I get these results. Why? 
![screenshot from 2017-05-10 18-24-36](https://cloud.githubusercontent.com/assets/16792714/25924061/c61a3c34-35ae-11e7-8418-39b7c9ffbef4.png)
The below is my code:

def get_generator_sampler():
	return torch.Tensor(np.random.randint(400, size=(1643, 68)))

def extract(v):
    return v.data.storage().tolist()

# GAN
g_input_size = 68  # Random noise dimension coming into generator, per output vector
g_hidden_size1 = 30  # Generator complexity
g_hidden_size2 = 14  # Generator complexity
g_output_size = 68  # size of generated output vector

d_input_size = 68  # cardinality of distributions
d_hidden_size1 = 30  # Discriminator complexity
d_hidden_size2 = 14  # Discriminator complexity
d_output_size = 1  # Single dimension for 'real' vs. 'fake'

d_learning_rate = 0.001
g_learning_rate = 0.001
optim_betas = (0.9, 0.999)

class Generator(nn.Module):
    def __init__(self,input_size,hidden_size1,hidden_size2,output_size):
        super(Generator,self).__init__()
        self.map1 = nn.Linear(input_size,hidden_size1)
        self.map2 = nn.Linear(hidden_size1,hidden_size2)
        self.map3 = nn.Linear(hidden_size2,output_size)

    def forward(self,X):
        wx_b1 = F.relu(self.map1(X))
        wx_b2 = F.sigmoid(self.map2(wx_b1))
        return torch.abs(self.map3(wx_b2))

class Discriminator(nn.Module):
    def __init__(self,input_size,hidden_size1,hidden_size2,output_size):
        super(Discriminator, self).__init__()
        self.map1 = nn.Linear(input_size, hidden_size1)
        self.map2 = nn.Linear(hidden_size1, hidden_size2)
        self.map3 = nn.Linear(hidden_size2, output_size)

    def forward(self,X):
        wx_b1 = F.relu(self.map1(X))
        wx_b2 = F.relu(self.map2(wx_b1))
        return F.sigmoid(self.map3(wx_b2))

G = Generator(input_size=g_input_size,hidden_size1=g_hidden_size1,hidden_size2=g_hidden_size2,output_size=g_output_size)
D = Discriminator(input_size=d_input_size,hidden_size1=d_hidden_size1,hidden_size2=d_hidden_size2,output_size=d_output_size)
criterion = nn.BCELoss()
d_optimizer = optim.Adam(D.parameters(), lr=d_learning_rate, betas=optim_betas)
g_optimizer = optim.SGD(G.parameters(), lr=g_learning_rate, momentum=0.9)

epoch = 0
num_epochs = 100000
for epoch in range(num_epochs):

    D.zero_grad()

    # 1. Train D on real
    d_real_data = Variable(data)
    d_output = D(d_real_data)
    d_real_error = criterion(d_output, Variable(torch.ones(len(d_output))))
    d_real_error.backward()

    # 2. Train D on fake
    z_samples = Variable(get_generator_sampler())
    f_output = D(G(z_samples))
    g_fake_error = criterion(f_output, Variable(torch.zeros(len(f_output))))
    g_fake_error.backward()

    d_optimizer.step()#Does update

    #3. Train G
    G.zero_grad()

    z_samples = Variable(get_generator_sampler())
    z_t_samples = G(z_samples)
    z_output = D(z_t_samples)
    z_fake_error = criterion(z_output, Variable(torch.ones(len(z_output))))#we want to fool D, so pretend it's all genuine
    z_fake_error.backward()
    g_optimizer.step()
